Data/Services/CustomerService.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Data.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Data.Services
+{
+    public class CustomerService : ICustomerService
+    {
+        private List<Customer> _customerList = new();
+
+        public void Create(Customer customer)
+        {
+            var checkingCustomerCpf = _customerList.Any(customerElement => customerElement.Cpf == customer.Cpf);
+
+            var checkingCustomerEmail = _customerList.Any(customerElement => customerElement.Email == customer.Email);
+
+            if (checkingCustomerCpf == true) throw new ArgumentException($"Customer for Cpf: {customer.Cpf} already exists!");
+
+            if (checkingCustomerEmail == true) throw new ArgumentException($"Customer for Email: {customer.Email} already exists!");
+
+            customer.Id = _customerList.LastOrDefault()?.Id + 1 ?? 1;
+            _customerList.Add(customer);
+        }
+
+        public IEnumerable<Customer> GetAll()
+        {
+            if (_customerList.Count == 0) throw new ArgumentException();
+            return _customerList;
+        }
+
+        public Customer GetById(long Id)
+        {
+            var customer = _customerList.FirstOrDefault(customer => customer.Id == Id);
+            if (customer != null) return customer;
+            throw new ArgumentException($"Customer not found for Id: {Id}");
```

Comment:
Mesma coisa mas de uma forma mais fácil de entender
```suggestion
            var customer = _customerList.FirstOrDefault(customer => customer.Id == Id)
                ?? throw new ArgumentException($"Customer not found for Id: {Id}");
            
            return customer;
```

Reply:
boaa